Bmx.Core/BmxCore.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Bmx.Core {
+	public class BmxCore {
+		private readonly IIdentityProvider _identityProvider;
+		public event PromptUserNameHandler PromptUserName;
+		public event PromptUserPasswordHandler PromptUserPassword;
+		public event PromptMfaTypeHandler PromptMfaType;
+		public event PromptMfaInputHandler PromptMfaInput;
+		public event PromptRoleSelectionHandler PromptRoleSelection;
+		public event PromptRoleSelectionHandler InformUnknownMfaTypesHandler;
+
+		private string _account;
+		private string _org;
+		private string _role;
+		private string _user;
+		private string _output;
+
+		public BmxCore( IIdentityProvider identityProvider ) {
+			_identityProvider = identityProvider;
+		}
+
+		public void Print( string account, string org, string role = null, string user = null,
+			string output = "powershell" ) {
+			_output = output;
+			DoCoreBmx( account, org, role, user ).Wait();
+		}
+
+		private async Task DoCoreBmx( string account, string org, string role = null, string user = null ) {
+			_account = account;
+			_org = org;
+			_role = role;
+			_user = user;
+
+			Debug.Assert( PromptUserName != null, nameof(PromptUserName) + " != null" );
+			Debug.Assert( PromptUserPassword != null, nameof(PromptUserPassword) + " != null" );
+			Debug.Assert( PromptMfaType != null, nameof(PromptMfaType) + " != null" );
+			Debug.Assert( PromptMfaInput != null, nameof(PromptMfaInput) + " != null" );
+			Debug.Assert( PromptRoleSelection != null, nameof(PromptRoleSelection) + " != null" );
+
+			if( user == null ) {
+				_user = PromptUserName( _identityProvider.Name );
+			}
+
+			// TODO: Handle case creds wrong / user mia
+			var mfaOptions =
+				await _identityProvider.Authenticate( _user, PromptUserPassword( _identityProvider.Name ) );
+
+			// TODO: Identify if non MFA use is possible with current setup for BMX, if so skip MFA steps
```

Comment:
Apologies in advance if I'm misunderstanding this comment.

If you're on VPN on a D2L computer, MFA is skipped with the Golang version of bmx.

Reply:
Got it, the no mfa use case I was not able to see in use. Will build out the path to skip MFA if its not required.